backend-agent/main.py

```diff
@@ -129,6 +140,116 @@ def check_health():
     return jsonify({'status': 'ok'})
 
 
+def allowed_file(filename):
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+@app.route('/api/upload-csv', methods=['POST'])
+def upload_csv():
+    if 'file' not in request.files:
+        return jsonify({'error': 'No file part'}), 400
+
+    file = request.files['file']
+    if file.filename == '':
+        return jsonify({'error': 'No file selected'}), 400
+
+    if file and allowed_file(file.filename):
+        filename = secure_filename(file.filename)
+        path = os.path.join(app.config['UPLOAD_FOLDER'], filename)
+        file.save(path)
+
+        # Optional: parse CSV immediately
+        try:
+            with open(path, newline='') as f:
+                reader = csv.DictReader(f)
+                data = list(reader)
+
+            return jsonify({'message': 'CSV uploaded successfully', 'data': data})
+
+        except Exception as e:
+            return jsonify({'error': f'Error reading CSV: {str(e)}'}), 500
+
+    return jsonify({'error': 'Invalid file type'}), 400
+
+
+# Endpoint to fetch all the vendors from the uploaded CSV
+@app.route('/api/vendors', methods=['GET'])
+def get_vendors():
+    # Check if CSV file exists
+    error_response = check_csv_exists('STARS_RESULTS.csv')
```

Comment:
please use a global variable for this name, so that we are ready to change it in case of need (it is referenced also at other lines as an hardcoded value)